core/server/worker/src/main/java/alluxio/worker/block/MonoBlockStore.java

```diff
@@ -301,44 +311,67 @@ public List<BlockStatus> load(List<Block> blocks, String tag, OptionalLong bandw
         blockWriter = mLocalBlockStore.createBlockWriter(sessionId, blockId);
       } catch (Exception e) {
         AlluxioRuntimeException exception = AlluxioRuntimeException.from(e);
-        errors.add(BlockStatus.newBuilder().setBlock(block)
-            .setCode(exception.getStatus().getCode().value()).setMessage(e.getMessage())
-            .setRetryable(exception.isRetryable()).build());
+        BlockStatus.Builder builder = BlockStatus.newBuilder().setBlock(block)
+            .setCode(exception.getStatus().getCode().value()).setRetryable(exception.isRetryable());
+        if (exception.getMessage() != null) {
+          builder.setMessage(exception.getMessage());
+        }
+        errors.add(builder.build());
+        if (hasTempBlockMeta(blockId)) {
+          try {
+            abortBlock(sessionId, blockId);
+          } catch (Exception ee) {
+            LOG.warn(format("fail to abort temp block %s when creating block for load request",
+                block.getBlockId()), ee);
+          }
+        }
         continue;
       }
-      CompletableFuture<BlockStatus> future = RetryUtils.retryCallable("read from ufs",
+      CompletableFuture<Void> future = RetryUtils.retryCallable("read from ufs",
               () -> manager.read(blockId, block.getOffsetInFile(), blockSize, block.getUfsPath(),
                   false, tag), new ExponentialBackoffRetry(1000, 5000, 5))
+          // use orTimeout in java 11
+          .applyToEither(timeoutAfter(LOAD_TIMEOUT, TimeUnit.MILLISECONDS), d -> d)
           .thenAcceptAsync(d -> blockWriter.append(ByteBuffer.wrap(d)),
               GrpcExecutors.BLOCK_WRITER_EXECUTOR)
           .thenRun(() -> {
             try {
               blockWriter.close();
             } catch (IOException e) {
-              abortBlock(sessionId, block.getBlockId());
               throw AlluxioRuntimeException.from(e);
             }
           })
-          .thenRun(() -> commitBlock(sessionId, blockId, false)).handle((res, throwable) -> {
-            if (throwable == null) {
-              return null;
-            } else {
+          .thenRun(() -> commitBlock(sessionId, blockId, false))
+          .exceptionally(t -> {
+            AlluxioRuntimeException exception = AlluxioRuntimeException.from(t.getCause());
```

Comment:
This same logic is copy-pasted twice. Why not have a `BlockStatus handleException(Exception e)` method (something along that line)